pkg/vpcmodel/grouping.go

```diff
@@ -165,6 +175,39 @@ func (g *groupedConnLine) getSrcOrDst(isSrc bool) EndpointElem {
 	return g.dst
 }
 
+// isOverApproximated() checks if the line was over approximated during the load balancer abstraction
+// it uses the lb AbstractionInfo that was kept during the approximation
+func (g *groupedConnLine) isOverApproximated() bool {
+	src, srcIsLb := g.src.(LoadBalancer)
+	dst, dstIsLb := g.dst.(LoadBalancer)
+	return srcIsLb && src.AbstractionInfo().hasMissingConnection(endpointElemResources(g.dst), false) ||
+		dstIsLb && dst.AbstractionInfo().hasMissingConnection(endpointElemResources(g.src), true)
+}
+
+// you might think that the following method should be part of EndpointElem interface.
+// however, there is no convenient way to do so (unless we add implementation for each VPCResource)
+// boo for golang
+func endpointElemResources(e EndpointElem) []VPCResourceIntf {
+	switch reflect.TypeOf(e).Elem() {
+	case reflect.TypeOf(groupedEndpointsElems{}):
+		elements := []EndpointElem(*e.(*groupedEndpointsElems))
+		r := make([]VPCResourceIntf, len(elements))
+		for i, e := range elements {
+			r[i] = e.(VPCResourceIntf)
+		}
+		return r
+	case reflect.TypeOf(groupedExternalNodes{}):
+		elements := []*ExternalNetwork(*e.(*groupedExternalNodes))
+		r := make([]VPCResourceIntf, len(elements))
+		for i, e := range elements {
+			r[i] = e
+		}
+		return r
+	default:
+		return []VPCResourceIntf{e.(VPCResourceIntf)}
+	}
+}
+
```

Comment:
func (n *VPCResource)endpointElemResources() []VPCResourceIntf {
	return []VPCResourceIntf{n.(VPCResourceIntf)}
}
does not compile, since VPCResource does not implement VPCResourceIntf

Reply:
I think this indicates a design problem.. can you think of some change of the current interfaces / adding another interface /...  to resolve this?

Reply:
I dont know
I have similar issues in the  drawio treenode structs.
there are bunch of methods there that does not have a receiver.

Reply:
opened issue #605